scripts/CheckCXXStdPutTime.cmake

```diff
@@ -0,0 +1,57 @@
+#
+# SRT - Secure, Reliable, Transport Copyright (c) 2022 Haivision Systems Inc.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+# Check for C++11 std::put_time().
+#
+# Sets:
+#   HAVE_CXX_STD_PUT_TIME
+
+include(CheckCSourceCompiles)
+
+function(CheckCXXStdPutTime)
+
+   unset(HAVE_CXX_STD_PUT_TIME CACHE)
+
+   set(CMAKE_TRY_COMPILE_TARGET_TYPE EXECUTABLE) # CMake 3.6
+
+   unset(CMAKE_REQUIRED_FLAGS)
+   unset(CMAKE_REQUIRED_LIBRARIES)
+   unset(CMAKE_REQUIRED_LINK_OPTIONS)
+
+   set(CheckCXXStdPutTime_CODE
+      "
+      #include <iostream>
+      #include <iomanip>
+      #include <ctime>
+      int main(void)
+      {
+         const int result = 0;
+         std::time_t t = std::time(nullptr);
+         std::tm tm = *std::localtime(&t);
+         std::cout
+            << std::put_time(&tm, \"%FT%T\")
+            << std::setfill('0')
+            << std::setw(6)
+            << std::endl;
+         return result;
+      }
+      "
+   )
+
+   # NOTE: Should we set -std or use the current compiler configuration.
+   #     It seems that the top level build does not track the compiler
+   #     in a consistent manner. So Maybe we need this?
+   set(CMAKE_REQUIRED_FLAGS "-std=c++11")
```

Comment:
Using particular C++ standard is quite a complicated thing; not sure what these "required" flags do, but whether C++11 standard is required to compile the C++ library, it's also checked elsewhere. This check upon confirmation should add its brick to requiring C++11 for compiling SRT library, but this is only one of the pieces and should be checked toghether with the others, for example, to check conflict with `--disable-c++11` flag.

Reply:
The `CMAKE_REQUIRED_FLAGS` are used by `check_cxx_source_compiles()` when trying to compile and is also of a course a local variable in the function. So it will not bleed over to the parent scope. You can use this method for instance to try and deduce what flags are needed to compile a bit of code. `check_cxx_source_compiles()` will use the current state of the `CMAKE_CXX_FLAGS` as well as any prior compiler configurations in addition to the `CMAKE_REQUIRED_*` , but since in the top level somettimes these are added to target configurations, I could not rely on them being set in the top level toolchain configuration even tho they could be set on some of the target stuff later. So this at least determines if this function is callable when -std=c++11 is set. Someone could add more conditions before calling `add_definitions(-DHAVE_CXX_STD_PUT_TIME=1)` should that become necessary.

This resolves some build breakage and should not break anything since the current source file that uses that definition is only built when `ENABLE_CXX11` is set.